src/utils/email.js

```diff
@@ -39,3 +40,44 @@ export function listEmailsInString (s) {
   }
   return result
 }
+
+/**
+ * @param {string} originalString
+ * @returns {{hashesFromOriginalString: HashedEmail[], stringWithoutRawEmails: string}}
+ */
+export function findAndReplaceRawEmails (originalString) {
+  if (containsEmailField(originalString)) {
+    return replaceEmailsWithHashes(originalString)
+  } else if (isEmail(originalString)) {
+    const hashes = hashEmail(originalString)
+    return {
+      stringWithoutRawEmails: hashes.md5,
+      hashesFromOriginalString: [hashes]
+    }
+  } else {
+    return {
+      stringWithoutRawEmails: originalString,
+      hashesFromOriginalString: []
+    }
+  }
+}
+
+/**
+ *
+ * @param originalString
+ * @returns {{hashesFromOriginalString: HashedEmail[], stringWithoutRawEmails: string}}
+ */
+export function replaceEmailsWithHashes (originalString) {
+  const emailsInString = listEmailsInString(originalString)
+  const hashes = []
+  for (let i = 0; i < emailsInString.length; i++) {
+    const email = emailsInString[i]
+    const emailHashes = hashEmail(email)
+    originalString = originalString.replace(email, emailHashes.md5)
```

Comment:
Lets' use another variable here.  Returning `stringWithoutRawEmails: originalString` looks strange at the first sight.